packages/react-api/src/api/model.js

```diff
@@ -98,14 +105,24 @@ export function executeModel(props) {
         }
       : undefined;
 
-    queryParams.spatialFilters = JSON.stringify(spatialFilters);
-    queryParams.spatialDataType = spatialDataType;
+    if (spatialFilters) queryParams.spatialFilters = JSON.stringify(spatialFilters);
+    if (spatialDataType) queryParams.spatialDataType = spatialDataType;
+    if (spatialDataColumn) queryParams.spatialDataColumn = spatialDataColumn;
+
     if (spatialDataType !== 'geo') {
       if (source.spatialFiltersResolution !== undefined) {
         queryParams.spatialFiltersResolution = source.spatialFiltersResolution;
       }
       queryParams.spatialFiltersMode = source.spatialFiltersMode || 'intersects';
     }
+
+    // We should retrieve the spatialDataColumn as part of the params.column
+    if (props.model === 'table') {
+      queryParams.params = JSON.stringify({
+        ...params,
+        column: [...params.column, spatialDataColumn]
```

Comment:
Very minor.
I would prefer special parameter that either hides (`hiddenColumns`) some column or fetches extra column (`invisiableColumns`) than magic that just adds geo column to fetched set.

App will have power to use whatever "extra" column it needs - even if current use case is to add exactly same column as `spatialDataColumn`

Reply:
I like it 🤔 

The only "problem" I see is that we'll have to expose or move somewhere the extraction of the `spatialDataColumn`, because it's the one we need to pass down as "hiddenColumns` or whatever name we decide to use, WDYT?

Reply:
Agree with @donmccurdy but it's something that we need to check in the server-side click initiative. Also, I'm worried if PS uses this widget and suddenly they will start receiving geometries. @jmgaya It would be possible to request the geometries as an optional parameter of the table widget?

Reply:
I think this `if` is going to be difficult to adapt when we integrate `carto-api-client`. The TableWidget has a `columns` parameter that you could use to add the `spatialDataColumn` and you would need an extra property to hide this column. I know that it requires more changes but I think that would be cleaner.

Reply:
> When a unique ID is available, perhaps fetching the entire geometry into the table widget is not necessary?

Yup, I think this makes total sense! However, it's not ready yet, but we should probably re-visit this once we merge this 👍

Reply:
> I think this if is going to be difficult to adapt when we integrate carto-api-client

Absolutely agree! I've tried to keep this as minimal as possible, but in terms of reusability, we can handle this in a different way, let me propose something else 👍